src/devtools/pachdev/actions.go

```diff
@@ -0,0 +1,571 @@
+func WithPachVersion(v string) Option {
+	return func(op *deployOp) error {
+		op.pachVersion = v
+		return nil
+	}
+}
+
+func WithObjectStorage(v string) Option {
+	// WithPachVersion is an option for NewDeployOp that sets the version of the
+	// Pachyderm instance that will be deployed.
+	return func(op *deployOp) error {
+		op.pachVersion = v
+		return nil
+	}
+}
+
+func NewDeployOp(verbose bool, opts ...Option) (*deployOp, error) {
+	result := &deployOp{
+		pachVersion:   "local",
+		objectStorage: "minio",
+	}
+	for _, op := range opts {
+		if err := op(result); err != nil {
+			return nil, err
+		}
+	}
+
+	return result, nil
+}
+
+// ctx and cancel are captured by result.Ctx(), and will not be initialized
+// if that is never called
+func (op *deployOp) Ctx() context.Context {
+	if op.ctx != nil {
+		return op.ctx
+	}
+	op.ctx, op.cancel = context.WithCancel(context.Background())
+	return op.ctx
+}
+
+func (op *deployOp) Cancel() {
+	if op.cancel != nil {
+		op.cancel()
+	}
+}
+
+// KubeClient is an accessor for op.kubeClient that lazily initializes the
+// field if it's not already set.  All access to op.kubeClient should be through
+// this method, even internally, in case it hasn't been initialized yet.
+func (op *deployOp) KubeClient() *kubernetes.Clientset {
+	if op.kubeClient != nil {
+		return op.kubeClient
+	}
+
+	// Load the Kubernetes configuration from the default location
+	config, err := clientcmd.BuildConfigFromFlags("", clientcmd.RecommendedHomeFile)
+	if err != nil {
+		op.err = errors.Wrap(err, "could not load kubernetes config")
+	}
+
+	op.kubeClient, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		op.err = errors.Wrap(err, "could not create kubernetes client")
+	}
+	return op.kubeClient
+}
+
+// PachClient is an accessor for op.pachClient that lazily initializes the
+// field if it's not already set.  All access to op.kubeClient should be through
+// this method, even internally, in case it hasn't been initialized yet.
+func (op *deployOp) PachClient() *client.APIClient {
+	if op.pachClient != nil {
+		return op.pachClient
+	}
+
+	pachctlCfg := &pachctl.Config{}
+	var err error
+	op.pachClient, err = pachctlCfg.NewOnUserMachine(op.Ctx(), false)
+	if err != nil {
+		op.err = errors.Wrap(err, "could not create Pachyderm client")
+	}
+	return op.pachClient
+}
+
+// KindProvider is an accessor for op.kindProvider that lazily initializes the
+// field if it's not already set.  All access to op.kindProvider should be through
+// this method, even internally, in case it hasn't been initialized yet.
+func (op *deployOp) KindProvider() *cluster.Provider {
+	if op.kindProvider != nil {
+		return op.kindProvider
+	}
+	var kindLogger log.Logger = log.NoopLogger{}
+	if verbose {
+		// TODO(msteffen) literally no idea what good UX is here. Ideally we
+		// would log to stdout instead of stderr, but I can't access
+		// kind/pgk/internal/cli to change the logging destination. Am I holding
+		// this wrong?
```

Comment:
I think we'll delegate to our own logger.  sigs.k8s.io/kind/pkg/log.Logger looks like a standard k8s logger.  For client-go, we just dump non-contextual logs from that logger to Zap like:

```go
klog.SetLogger(zapr.NewLogger(l.Named("client-go")))
```

(zapr provides a klog/v2 fascade on top of zap basically.)

In this case, you can write a wrapper in `log` to return a zapr log (API should be `log.NewZapr(ctx)`) and then use that inside kind.  We do this for logrus, stdlog (i.e. "log" in the std library), etc. throughout the codebase, as many libraries use loggers other than the pachyderm native logger ;)